nixos/modules/services/network-filesystems/ipfs.nix

```
@@ -199,13 +137,21 @@ in {
         example = 64*1024;
       };
 
+      startWhenNeeded = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to use socket activation to strt IPFS when needed.";
```

```suggestion
        description = "Whether to use socket activation to start IPFS when needed.";
```

I also assume once our `switch-to-configuration` logic becomes a bit smarter w.r.t. handling `.socket` files, we could also do this by default.